api/handlers_spaces.go

```diff
@@ -108,43 +186,54 @@ func (s *server) SpaceGetHandler(w http.ResponseWriter, r *http.Request) {
 			aws.String("USAGE_QUANTITY"),
 		},
 		TimePeriod: &costexplorer.DateInterval{
-			End:   aws.String(fmt.Sprintf("%d-%02d-%02d", y, m, d)),
-			Start: aws.String(fmt.Sprintf("%d-%02d-01", y, m)),
+			Start: aws.String(start),
+			End:   aws.String(end),
 		},
 	}
 
-	// the object is not found in the cache, call AWS cost-explorer and set cache
-	var out []*costexplorer.ResultByTime
-	c, expire, ok := resultCache.GetWithExpiration(spaceID)
-	if !ok || c == nil {
-		log.Debugf("cache empty for org, space: %s, %s, calling cost-explorer", Org, spaceID)
-		// call cost-explorer
-		var err error
-		out, err = ceService.GetCostAndUsage(r.Context(), &input)
+	// if time did not parse correctly, don't call AWS cost explorer
+	if timeValidity == true {
```

Comment:
i still don't think we need this boolean?  (also `if timeValidity {`)

Reply:
gone